src/service/app_state.py

```diff
@@ -58,6 +62,18 @@ async def _build_storage(cfg: CollectionsServiceConfig, data_products: list[Data
         raise
 
 
+async def _get_workspace_url(cfg: CollectionsServiceConfig) -> str:
+    if cfg.dont_connect_to_external_services:
+        return False
+    ws = Workspace(cfg.workspace_url)
```

Comment:
OK. But just in case, I would move `ws = Workspace(cfg.workspace_url)` to the try-except clause below.

Reply:
The only error that could occur here is a url parse error, for which the catch / rethrow isn't really necessary as the error message should provide enough info, but it doesn't hurt either, so fair enough